objects/src/notes/inputs.rs

```diff
@@ -55,12 +59,20 @@ impl NoteInputs {
         &self.inputs
     }
 
-    /// Returns a hash digest of the inputs. Computed as a linear hash of the inputs.
+    /// Returns a commitment to these inputs.
     pub fn hash(&self) -> Digest {
-        self.hash
+        *self.hash.get_or_init(|| Hasher::hash_elements(&self.inputs))
     }
 }
 
+impl PartialEq for NoteInputs {
+    fn eq(&self, other: &Self) -> bool {
+        self.inputs == other.inputs
```

Comment:
Similar to previous comment, we could destructure `self` here to clearly state our assumptions about the current structure of `NoteInputs`

Reply:
Good idea! Updated.